src/pybind/mgr/prometheus/module.py

```diff
@@ -104,7 +104,7 @@ def health_status_to_number(status):
                        'ceph_version')
 
 DISK_OCCUPATION = ('ceph_daemon', 'device', 'db_device',
-                   'wal_device', 'instance')
```

Comment:
Can we create a new metric for this instead? (especially if we want to back-port this) We wanted to make the metrics more stable once we released Nautilus and given how widely this metric is used, this would cause at least temporary matrching issues (upon upgrade).

Reply:
puttng the device information with the other device information seemed like the logical place to me. the metric name is stable, I'm just seeking to add labels. existing queries won't reference it, so I'm not clear on what the problem is. I'm missing something - sorry! Can you explain more as to why the addition of a label requires a whole new metric, and what the matching issues will be?

Reply:
Once you upgrade, you will be getting dual metrics for ~5 minutes (until the old metrics without the new labels time out). There is also a chance to break matching on this metric as there will be new labels that the query might not be able to handle. IIRC, this metric is widely used for matching so I wouldn't be surprised if this change would break a couple of queries.

Reply:
Agree that after upgrade the metric will exist with old labels and new labels for a period. But any queries we have match on the current labels, which are unaffected by this change - so I fail to see how queries break when new labels are added that are not referenced at all by the existing code, and the label names are not changing.

Reply:
You can use `excluding` for labels when matching and that would cause issues. I know we have had an issue with this in the past (not exactly sure for which metric though). Even if we don't use it in our metrics, other people might and we should try to provide stable metrics at this point.

Reply:
I don't agree that we should introduce new metrics on the off chance that someone, somewhere decides to create dashboards/alerts - surely we can only support what we provide. There's a plethora of things that users can do but we can only support what we provide right?

Reply:
Well, then let @jan--f decide. I know we wanted to stabilise the metrics and this might break the stable interface. @jan--f What do you think?

Reply:
I agree with @pcuzner here. Its the logical place and we can't add new metrics or not develop this further for fear of _maybe_ breaking some queries.

Its probably a good idea to add a release note entry though.